### PR TITLE
#530 Clarification around datacontenttype

### PR DESCRIPTION
**Pull request by gunnarmorling**
Related to: https://github.com/cloudevents/spec/issues/530

Fixes: #530

### CODE REVIEW

**Comment by duglin** [comment]
@gunnarmorling can you add a comment to this one? If nothing else at least a `Fixes #...` so we know the issue it's related to?

**Comment by duglin** [comment]
related to: https://github.com/cloudevents/spec/issues/530

**Comment by duglin** [comment]
@gunnarmorling @bluemonk3y where are we on this?

Also - rebase needed

**Comment by bluemonk3y** [comment]
@duglin  - I left a comment on #530

File: kafka-transport-binding.md [hunk 1]
```diff
@@ -102,10 +100,10 @@ here.
 
 The receiver of the event can distinguish between the two content modes by
 inspecting the `ce_datacontenttype` [Header][Kafka-Message-Header] of the
```
Comment (duglin) [inline kafka-transport-binding.md]: shouldn't this be `content-type` ?
@clemensv

File: kafka-transport-binding.md [hunk 2]
```diff
@@ -102,10 +100,10 @@ here.
 
 The receiver of the event can distinguish between the two content modes by
 inspecting the `ce_datacontenttype` [Header][Kafka-Message-Header] of the 
-Kafka message. If the value is prefixed with the CloudEvents media type
-`application/cloudevents`, indicating the use of a known
-[event format](#14-event-formats), the receiver uses *structured* mode, otherwise
-it defaults to *binary* mode.
+Kafka message. If no value is given, or if the value is prefixed with the
```
Comment (duglin) [inline kafka-transport-binding.md]: If I'm correct, and it should be `content-type` above, then I think "no value" should default to binary not structured, and then the original text would make more sense.

File: kafka-transport-binding.md (hunk 1 above)
Comment (duglin) [reply kafka-transport-binding.md]: @gunnarmorling if we fix this line would that address your confusion mentioned in #530 ?
I think the rest of the text in the spec is correct and doesn't need to be updated (except for the typo for `type`).

Comment (gunnarmorling) [reply kafka-transport-binding.md]: Yes, using `content-type` consistently as the name of the header property helps 👍

File: kafka-transport-binding.md (hunk 2 above)
Comment (gunnarmorling) [reply kafka-transport-binding.md]: How will this work with Apache Kafka < 0.11 then though (also commented about this on your new PR)? Headers cannot be used there, so all the other `ce_...` attributes couldn't be given. Am I missing something?

**Comment by gunnarmorling** [comment]
Superseded by #537; leaving it open until the pending discussions have been resolved.

**Comment by gunnarmorling** [comment]
Closing, #537 addresses these concerns.

File: kafka-transport-binding.md (hunk 2 above)
Comment (alanconway) [reply kafka-transport-binding.md]: MQTT <= 3.1 >= 4 has a similar issue. The approach there was MUST use JSON format and not support anything else on older MQTT versions.

**Pull request closed without merging**